### PR TITLE
[ticket/16526] Correctly handle ACP CSV settings

### PR DESCRIPTION
**Pull request by rxu**
Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

<a href="https://tracker.phpbb.com/browse/PHPBB3-16526">PHPBB3-16526</a>.

### CODE REVIEW

**Comment by 3D-I** [comment]
I thought it was fixed with https://github.com/phpbb/phpbb/pull/5935

I see this one is doing some extras :+1:

**Comment by rxu** [comment]
Thanks, I didn't notice this change for some reason, so reverted back the change for `s9e/factory` here.

**Comment by 3D-I** [comment]
> 
> 
> Thanks, I didn't notice this change because was doing testing on released versions, so reverted back the change for `s9e/factory` here.

My fix made sure that whatever is entered in the ACP scheme links is filtered so it excludes future mistakes but obviously your PR is much better articulated.

File: phpBB/includes/functions_acp.php [hunk 1]
```diff
@@ -453,8 +453,20 @@ function validate_config_vars($config_vars, &$cfg_array, &$error)
 					$error[] = $language->lang('URL_INVALID', $language->lang($config_definition['lang']));
 				}
 
+				goto validate_string;
```
Comment (marc1706) [inline phpBB/includes/functions_acp.php:456]: Goto should not be used in accordance with the coding guidelines.

**Pull request merged**